src/doc/src/reference/external-tools.md

```diff
@@ -46,6 +46,10 @@ The `package_id` field is a unique identifier for referring to the package, and
 as the `--package` argument to many commands. The syntax grammar can be found in
 chapter [Package ID Specifications].
 
+> **Note:** Not all output is guaranteed to be in JSON format with the `--message-format=json` option.
+> For example, output from procedural macros and logs from external tools may appear in plain text.
+> Tools should only interpret a line as JSON if it starts with `{`.
+
```

Comment:
I think I'd prefer we don't bless mixed output as much as this makes it sound like it.

Maybe instead
```suggestion
> **Note:** `--message-format=json` only controls Cargo and Rustc's output.  This cannot control the output of other tools, e.g. `cargo run --message-format=json`.  A possible workaround in these situations is to only interpret a line as JSON if it starts with `{`.

```

Reply:
Applied your suggestion with a mention of proc macros. I think it is still worth since proc macros are not seen as “other tools” and people may not realize immediately.

Reply:
Yeah, they are a nother of those "outside of the control of cargo/rustc" things that are worth calling out